server/src/main/java/io/crate/planner/operators/SubQueryResults.java

```diff
@@ -96,4 +96,27 @@ public Void visitOuterColumn(OuterColumn outerColumn, Void context) {
     public void bindOuterColumnInputRow(Row inputRow) {
         this.inputRow = inputRow;
     }
+
+    /**
+     * Combines correlated join related fields and regular sub-select related fields
+     * by getting another object's value if own value is not set/empty.
+     *
+     * Needed to get an object preserving all required payload to execute a statement,
+     * which mixes correlated subquery and sub-select.
+     */
+    public SubQueryResults combineCorrelationAndSubQuery(SubQueryResults subQueryResults) {
```

Comment:
I'd prefer to declare that static and just call it `merge`. I'd also remove the comment as the method name in combination with the arguments should describe it already well.
```suggestion
    public static SubQueryResults merge(SubQueryResults subQueryResults1, SubQueryResults subqueryResults2) {
```

Reply:
Or lets keep it non-static but call it `forCorrelation`? 
So it would align with the existing `forCorrelation` method..

Reply:
I changed to merge - `forCorrelation` might be misleading as it's not only about it, it's more about mixed `SubQueryResults` users in the same scope